### DISCUSSION TITLE
Replace Inverter - Offset Data

### DISCUSSION THREAD

**Discussion by Demel75** [SBFspot/SBFspot#691] [Q&A] [answered]
I am just changed an inverter from SB 3600TL-21 to STP10.0-3SE-40. Comunication works fine and data ist uploaded to mysql an also pvoutput.org transfer works.

Unfortunately the new values start again at 0 kWh from the inverter. So data isn't stored right e.g. on pvoutput.org.

Is it possible to define an offset to go on with the data as a complete evaluation.

Now it look like this e.g. in CSV:
```
05/02/2024 15:55:00;29483,854;0,084
05/02/2024 16:25:00;22,603;0,060
```
First line old inverter, second line new inverter.

If I can use for e.g. a replace_offest=29483.854

the lines could look like this:
```
05/02/2024 15:55:00;29483,854;0,084
05/02/2024 16:25:00;29506,457;0,060
```

Is this possible? Or any other ideas for replacement and data consistency?

Thanks for help.

**Comment by sbf-** [comment]
Upload data is taken from SQL database, so you can change the data before upload, but that will lead to possible errors if that script fails...
I think it's better to start a new system on pvoutput, because it is... a new system after all

**Comment by Demel75** [comment, marked as answer]
It's only a new inverter, no new system. I like to have my history for my home.


Maybe we can edit _ArchData.cpp_ like this:

```
inverters[inv]->dayData[idx].totalWh = totalWh;
inverters[inv]->dayData[idx].totalWh += replace_offset;
```

**Comment by sbf-** [comment]
So you changed 3.6 with 10.0 inverter and same solar panels?

> Maybe we can edit _ArchData.cpp_

That's possible, but when you upgrade SBFspot you'll need to redo this every time...

You can also modify the vwPvoData view:

```
SELECT dd.Timestamp,
              dd.Name,
              dd.Type,
              dd.Serial,
              dd.TotalYield+MyOffset AS V1,
              dd.Power AS V2,
....
```

**Comment by Demel75** [comment]
Yes. The new one was cheap and has emergency power.
Thanks for help.

**Comment by silliewimons** [comment]
to demel75,     congratulations with your STP10.0-3SE-40     (read comment at end)

I assume that PVoutput system can handle more hen one inverter as part of your "home-system",                    
if that solution does not fit your needs  AND depending on which values to export as continuation-from-> SB3600TL

by using SunnyExplorer, you can modify - the cumul-counter for production  (value within your new inverter)
     you can modify / increase the actual value with eg. "29483.854", 
 !!! this method  SHOULD BE USED  when a failing inverter is replaced by another one   to  provide continued  counters 
     be aware - one MUST  execute this operation as part of "swapping inverters"
     this method **is NOT recommended**  AFTER the swapped inverter is in operation because
              one needs to backwards correct several values  towards "PVoutput" and  . . . .

to sbf,   this is another example where "plant" covers more then one 'equipment', in this case a replacement of . . . 

remarks on STP10.0-3SE-40 - this type of inverter includes 'SMA Smart Screen' - a build-in webservice
     which offers more options then SunnyExplorer, - after inititation "first-usage-setup" - one gets access to a detailed history

kr sillie

**Reply by Demel75** [reply]
Hi sillie, thanks for your reply.

Yes, I set allready that "OFFSET" value  in the SMA inverter. But SBFSpot don't see that value with offset (SMA: "PvGen.PvWhSet") - is seems not the same as ETotal. Or have I configure something different on SBFspot?

![image](https://github.com/SBFspot/SBFspot/assets/18624555/08e28d31-747e-4a1a-80df-e2f618c3cf64)

normal SBFspot:
```
Energy Production:
        EToday: 0.112kWh
        ETotal: 28.229kWh
        Operation Time: 0.00h
        Feed-In Time  : 0.00h

12/02/2024 09:10:00 : 28.119kWh - 12.000W
12/02/2024 09:15:00 : 28.121kWh - 24.000W
12/02/2024 09:20:00 : 28.124kWh - 36.000W
12/02/2024 09:25:00 : 28.128kWh - 48.000W
12/02/2024 09:30:00 : 28.134kWh - 72.000W
12/02/2024 09:35:00 : 28.151kWh - 204.000W
12/02/2024 09:40:00 : 28.173kWh - 264.000W
12/02/2024 09:45:00 : 28.199kWh - 312.000W
```

my moded SBFSpot: 
```
Energy Production:
        EToday: 0.121kWh
        ETotal: 28.238kWh
        Operation Time: 0.00h
        Feed-In Time  : 0.00h

12/02/2024 09:10:00 : 29511.973kWh - 12.000W
12/02/2024 09:15:00 : 29511.975kWh - 24.000W
12/02/2024 09:20:00 : 29511.978kWh - 36.000W
12/02/2024 09:25:00 : 29511.982kWh - 48.000W
12/02/2024 09:30:00 : 29511.988kWh - 72.000W
12/02/2024 09:35:00 : 29512.005kWh - 204.000W
12/02/2024 09:40:00 : 29512.027kWh - 264.000W
12/02/2024 09:45:00 : 29512.053kWh - 312.000W
```

**Comment by silliewimons** [comment]
dear demel75, remember that as a result of swapping your SB with STP-SE the descriptions/terminology did change
 - example - Etotal with SB == production by PV-solar     with SmartEnergy this has a different 'idty'

allow me to take some reflection-time about this point,
      -SB-term  Etotal = PVproduction  while STP-SE-term  Etotal = energy-flow TO local-grid

kr sillie
I include a document with  SMA-drawings on this subject
[wsi11a681.SMA-SXX-basic-config.docx](https://github.com/SBFspot/SBFspot/files/14269342/wsi11a681.SMA-SXX-basic-config.docx)

**Comment by silliewimons** [comment]
basic reflections:    
 - with SunnyBoy the PVproduction(DC) was NEAR equal to Etotal(AC) -- DELTA was efficiency of your inverter 
    with STP-SE  the energy flows are different - because energy flow TO / FROM  the battery  is added
 !! with PVoutput one can 'report' various measurements about HOME-energy-flow: -"local-consumption"; -Export-TO-grid; -Import-FROM-grid   
 !! the options/reported values depend on your account with PVoutput  AND  settings via SBF-Upload-cfg

REMARK details for "Grid-In&Out / LocalConsump" is not available from STP-SE, it requires a measurement at the Grid-connecton-point

Summary: -Energy-flow(AC) (in/out) from STP#SE = 'Metering.TotWhOut'  == ETotal 
                 -PVgeneration(DC) = 'PvGen.PvWh'
                  the difference  is  sum of: -"BatChrg"; -"BatDsch"; -"loss"&"efficiency"
           
kind regards  sillie

**Comment by silliewimons** [comment]
By using 'Offset' - it seems that  one  'add'-'substract' - offset from  values in history,  this is different from "changing" the counter
in below image, you see  E-total !!!always possitive  (dutch-name) &  the option to apply 'offset' value-range -4000.000 <-> +4000.000
offset   is  a static value   and is part of the "fixed-values"     E-total  is a counter
![image](https://github.com/SBFspot/SBFspot/assets/34144830/31cd6e6d-092d-41b5-98db-24393759ec31)
